resource/generation/client.go

```diff
@@ -600,8 +407,8 @@ func formatInterfaceTypes(types []string) string {
 
 func formatResourceInterfaceTypes(resources []resourceInfo) string {
 	names := make([]string, len(resources))
-	for i, resource := range resources {
-		names[i] = resource.Name()
+	for i := range resources {
+		names[i] = resources[i].Name()
 	}
```

Comment:
This is a bit of an anti-pattern. We should use append unless there is real utility in using an index.


```suggestion
	names := make([]string, 0, len(resources))
	for _, resource := range resources {
		names = append(names, resource.Name())
	}
```

Reply:
@jwatson-ccc I refactored this when fixing linter errors warning about shadowing the package name `resource`. Using `make(T, len)` instead of `make(T, 0, capacity)` always shoots me in the foot and I try to avoid it. In this case the existing `names[i]` indexing prevented that, and I never noticed it.